src/main/java/seedu/address/model/attendance/AttendanceRecord.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Represents the attendance of students in a single lesson class.
+ * Guarantees Immutability.
+ */
+public class AttendanceRecord {
+
+    private final Map<UUID, Attendance> record;
+
+    public AttendanceRecord() {
+        this.record = new HashMap<>();
+    }
+
+    /**
+     * Overloaded constructor method.
+     * Requires record to be non null.
+     */
+    public AttendanceRecord(Map<UUID, Attendance> record) {
+        requireNonNull(record);
+
+        this.record = record;
+    }
+
+    public Map<UUID, Attendance> getAttendanceRecord() {
+        return Collections.unmodifiableMap(record);
+    }
+
+    /**
+     * Returns a deep copied {@code AtendanceRecord}.
+     */
+    public AttendanceRecord deepCopy() {
+        return new AttendanceRecord(
+                deepCopyMap(record, uuidKey -> uuidKey, value -> value.deepCopy()));
+    }
+
+    /**
+     * Gets the {@code Attendance} of the given {@code UUID}.
+     */
+    public Attendance getAttendance(UUID uuid) throws AttendanceNotFoundException {
+        requireNonNull(uuid);
+
+        if (!record.containsKey(uuid)) {
+            throw new AttendanceNotFoundException();
+        }
+
+        return record.get(uuid);
+    }
+
+    /**
+     * Adds a {@code UUID} and {@code Attendance} to the map.
+     */
+    public AttendanceRecord addAttendance(UUID uuid, Attendance attendance)
+            throws DuplicateAttendanceException {
+        requireAllNonNull(uuid, attendance);
+
+        if (record.containsKey(uuid)) {
+            throw new DuplicateAttendanceException();
+        }
+
+        Map<UUID, Attendance> copiedRecord =
+                deepCopyMap(record, uuidKey -> uuidKey, value -> value.deepCopy());
+        copiedRecord.put(uuid, attendance);
+        return new AttendanceRecord(copiedRecord);
+    }
+
+    /**
+     * Edits the {@code Attendance} of a given {@code UUID} in the map.
+     */
+    public AttendanceRecord editAttendance(UUID uuid, Attendance editedAttendance)
+            throws AttendanceNotFoundException {
+        requireAllNonNull(uuid, editedAttendance);
+
+        if (!record.containsKey(uuid)) {
+            throw new AttendanceNotFoundException();
+        }
+
+        Map<UUID, Attendance> copiedRecord =
+                deepCopyMap(record, uuidKey -> uuidKey, value -> value.deepCopy());
+        copiedRecord.put(uuid, editedAttendance);
+        return new AttendanceRecord(copiedRecord);
+    }
+
+    /**
+     * Deletes the given {@code UUID} entry from the map.
+     */
+    public AttendanceRecord deleteAttendance(UUID uuid) throws AttendanceNotFoundException {
+        requireNonNull(uuid);
+
+        if (!record.containsKey(uuid)) {
+            throw new AttendanceNotFoundException();
+        }
+
+        Map<UUID, Attendance> copiedRecord =
+                deepCopyMap(record, uuidKey -> uuidKey, value -> value.deepCopy());
+        copiedRecord.remove(uuid);
+        return new AttendanceRecord(copiedRecord);
+    }
```

Comment:
I think it would be best for these to be left to be handled by the logic part of the house for consistency with the rest of the model components. It would be sufficient to provide getter methods and a suitable constructor.